StreetCare/StreetCare/Views/Screens/AddHelpRequestForm.swift

```diff
@@ -0,0 +1,449 @@
+                            saveToFirestore()
+                        } else {
+                            alertMessage = "Please fill in all required fields."
+                            showAlert = true
+                        }
+                    }
+                    .alert(isPresented: $showAlert) {
+                        Alert(
+                            title: Text("Form Submission"),
+                            message: Text(alertMessage),
+                            dismissButton: .default(Text("OK")) {
+                                if alertMessage == "Event saved successfully!" {
+                                    presentationMode.wrappedValue.dismiss()
+                                }
+                            }
+                        )
+                    }
+                Spacer() // Push content to center
+            }
+
+            // Discard Button
+            HStack {
+                Spacer() // Push content to center
+                NavLinkButton(title: NSLocalizedString("discardButtonTitle", comment: ""), width: 300, secondaryButton: true)
+                    .padding()
+                    .background(Color.clear)
+                    .onTapGesture {
+                        discardAction()
+                    }
+                Spacer() // Push content to center
+            }
+        }
+        .frame(maxWidth: .infinity) // Ensure the VStack takes full width for proper centering
+    }
+}
+
+struct FormHeaderView: View {
+    var body: some View {
+        VStack(alignment: .leading, spacing: 20) {
+            Text("🙌 Need extra help?\nLet us spread the word!")
+                .font(.largeTitle)
+                .fontWeight(.bold)
+                .multilineTextAlignment(.leading)
+
+            Text("Publish a request to Community Hub")
+                .font(.title2)
+                .fontWeight(.semibold)
+
+            Text("If this homeless person requires further assistance, please provide additional details so we can share with the community and rally support to help them.")
+                .font(.subheadline)
+                .foregroundColor(.gray)
+        }
+        .frame(maxWidth: .infinity, alignment: .leading)
+        .padding(.top, 0) // Remove or reduce top padding
+    }
+}
+
+/// Text Field with Character Limit
+struct TextFieldWithLimit: View {
+    var title: String
+    var placeholder: String
+    @Binding var text: String
+    var limit: Int
+
+    var body: some View {
+        VStack(alignment: .leading) {
+            Text(title).font(.headline)
+            
+            // TextField with border
+            TextField(placeholder, text: Binding(
+                get: { text },
+                set: { text = String($0.prefix(limit)) }
+            ))
+            .frame(height:50)
+            .textFieldStyle(PlainTextFieldStyle()) // Use PlainTextFieldStyle to customize the border
+            .padding() // Padding inside the TextField for spacing
```

Comment:
I noticed that there are a lot of comments throughout the code. While comments can be helpful for explaining complex logic, having too many comments might reduce readability and clutter the code. Ideally, the code itself should be self-explanatory where possible (through clear variable names, functions, and structure).

Perhaps we could remove or rephrase comments that are redundant or overly detailed, keeping only those that add real value or explain non-obvious logic.